test/sentry/check_in_id_mappings_test.exs

```diff
@@ -0,0 +1,30 @@
+defmodule Sentry.CheckInIDMappingsTest do
+  # This is not async because it tests a singleton (the CheckInIDMappings GenServer).
+  use Sentry.Case, async: false
+
+  alias Sentry.Integrations.CheckInIDMappings
+  @table :sentry_cron_mappings
+
+  describe "lookup_or_insert_new/1" do
+    test "works correctly" do
+      cron_key = "quantum_123"
+
+      child_spec = %{
+        id: TestMappings,
+        start:
+          {CheckInIDMappings, :start_link, [[max_expected_check_in_time: 0, name: TestMappings]]}
+      }
+
+      pid = start_supervised!(child_spec)
+
+      CheckInIDMappings.lookup_or_insert_new(cron_key)
+      assert :ets.lookup(@table, cron_key) != []
+
+      Process.sleep(5)
```

Comment:
Why sleep here?

Reply:
We need to sleep to allow the `:sweep` message to be processed by the target process so that the final assertion accurately reflects the updated state of the ETS table. Otherwise, it returns the key/value pair before being deleted.

Reply:
If you call `:sys.get_state(pid)`, you are making a sync call to the GenServer. Since messages are ordered, you are guaranteed that by the time the GenServer replies to the `:sys.get_state/1` call then the `:sweep` message will have been processed. So, you can remove this 🙃

Reply:
Maybe I'm missing something? When I remove the sleep it returns the pair and the final assert fails

```   test/sentry/check_in_id_mappings_test.exs:9
     Assertion with == failed
     code:  assert :ets.lookup(@table, cron_key) == []
     left:  [{"quantum_123", "ec2ded4d0c1a43b580d2e567396e3bee", 1723541054291}]
     right: []
     stacktrace:
       test/sentry/check_in_id_mappings_test.exs:27: (test)
```

Reply:
Ok no rush I can debug this locally.